tests/foreman/api/test_syncplan.py

```diff
@@ -156,6 +157,8 @@ def test_positive_create_enabled_disabled(module_org, enabled):
     sync_plan = entities.SyncPlan(enabled=enabled, organization=module_org).create()
     sync_plan = sync_plan.read()
     assert sync_plan.enabled == enabled
+    # disable sync plan after test
+    disable_syncplan(sync_plan)
```

Comment:
So, consider what happens if the assert fails, or if Nailgun throws an exception during the read - this line will never get executed, so the syncplan will stay enabled.

In order to make this a stronger 'cleanup, I would move these calls into the test teardown by using `request.addfinalizer`.

Here this would look like: 

```
def test_positive_create_enabled_disabled(module_org, enabled, request, target_sat):
    sync_plan = target_sat.api.SyncPlan(enabled=enabled, organization=module_org).create()
    request.addfinalizer(lambda: disable_syncplan(sync_plan))
    sync_plan = sync_plan.read()
    assert sync_plan.enabled == enabled
```

Reply:
This applies to anywhere that you're directly calling `disable_syncplan` within the test.

It should be called via `request.addfinalizer()` right after it gets created.  The function won't actually execute until pytest is in the teardown phase.